aws/resource_aws_dlm_lifecycle_policy_test.go

```diff
@@ -0,0 +1,292 @@
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.resource_types.0", "VOLUME"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.schedule.0.name", "tf-acc-full"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.schedule.0.create_rule.0.interval", "12"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.schedule.0.create_rule.0.interval_unit", "HOURS"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.schedule.0.create_rule.0.times.0", "21:42"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.schedule.0.retain_rule.0.count", "10"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.schedule.0.tags_to_add.tf-acc-test-added", "full"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.target_tags.tf-acc-test", "full"),
+				),
+			},
+			{
+				Config: dlmLifecyclePolicyFullUpdateConfig(),
+				Check: resource.ComposeTestCheckFunc(
+					checkDlmLifecyclePolicyExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "description", "tf-acc-full-updated"),
+					resource.TestCheckResourceAttrSet(resourceName, "execution_role_arn"),
+					resource.TestCheckResourceAttr(resourceName, "state", "DISABLED"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.resource_types.0", "VOLUME"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.schedule.0.name", "tf-acc-full-updated"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.schedule.0.create_rule.0.interval", "24"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.schedule.0.create_rule.0.interval_unit", "HOURS"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.schedule.0.create_rule.0.times.0", "09:42"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.schedule.0.retain_rule.0.count", "100"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.schedule.0.tags_to_add.tf-acc-test-added", "full-updated"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.target_tags.tf-acc-test", "full-updated"),
+				),
+			},
+		},
+	})
+}
+
+func dlmLifecyclePolicyDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).dlmconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_dlm_lifecycle_policy" {
+			continue
+		}
+
+		input := dlm.GetLifecyclePolicyInput{
+			PolicyId: aws.String(rs.Primary.Attributes["name"]),
+		}
+
+		out, err := conn.GetLifecyclePolicy(&input)
+
+		if err != nil {
+			if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "ResourceNotFoundException" {
+				return nil
+			}
+			return err
+		}
+
+		if out.Policy != nil {
+			return fmt.Errorf("DLM lifecycle policy still exists: %#v", out)
+		}
+	}
+
+	return nil
+}
+
+func checkDlmLifecyclePolicyExists(name string) resource.TestCheckFunc {
```

Comment:
This function should implement a similar (but opposite) check to the CheckDestroy, where it verifies that the expected resource does actually exist in the API. Admittedly, this is a little buried in the provider development documentation, here: https://www.terraform.io/docs/extend/testing/acceptance-tests/teststep.html#custom-check-functions

```go
return func(s *terraform.State) error {
  rs, ok := s.RootModule().Resources[name]

  if !ok {
    return fmt.Errorf("Not found: %s", name)
  }

  if rs.Primary.ID == "" {
    return fmt.Errorf("Resource %s ID not found", name)
  }

  input := dlm.GetLifecyclePolicyInput{
    PolicyId: aws.String(rs.Primary.ID),
  }

  out, err := conn.GetLifecyclePolicy(&input)

  if err != nil {
    return fmt.Errorf("error getting DLM Lifecycle Policy (%s): %s", rs.Primary.ID, err)
  }

   return nil
}
```